File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/mission/model/Mission.java
package com.study.webflux.rag.domain.mission.model;

public record Mission(
	MissionId missionId,
	MissionType type,
	String name,
	String description,
	long rewardAmount,
	boolean repeatable
) {
	public Mission {
		if (missionId == null) {
			throw new IllegalArgumentException("missionId cannot be null");
		}
		if (type == null) {
			throw new IllegalArgumentException("missionType cannot be null");
		}
		if (name == null || name.isBlank()) {
			throw new IllegalArgumentException("mission name cannot be blank");
		}
		if (rewardAmount <= 0) {
			throw new IllegalArgumentException("rewardAmount must be positive");
		}
	}

	public static Mission create(MissionType type, String name, String description, long rewardAmount,
		boolean repeatable) {
		return new Mission(MissionId.generate(), type, name, description, rewardAmount, repeatable);
	}
}

File: backend/domain/src/main/kotlin/com/miyou/app/domain/mission/model/Mission.kt
package com.miyou.app.domain.mission.model

data class Mission(
    val missionId: MissionId,
    val type: MissionType,
    val name: String,
    val description: String?,
    val rewardAmount: Long,
    val repeatable: Boolean,
) {
    init {
        require(name.isNotBlank()) { "미션 이름은 비어 있을 수 없습니다." }
        require(rewardAmount > 0) { "보상 금액은 양수여야 합니다." }
    }

    companion object {
        @JvmStatic
        fun create(
            type: MissionType,
            name: String,
            description: String?,
            rewardAmount: Long,
            repeatable: Boolean,
        ): Mission = Mission(MissionId.generate(), type, name, description, rewardAmount, repeatable)
    }

    fun missionId(): MissionId = missionId

    fun type(): MissionType = type

    fun name(): String = name

    fun description(): String? = description

    fun rewardAmount(): Long = rewardAmount

    fun repeatable(): Boolean = repeatable
}
